### PR TITLE
sql: add index direction to SHOW CREATE TABLE

### LINKED ISSUE #13923: sql: SHOW CREATE TABLE does not report index directions
```
root@:26257/t> create table ax(a int, b int, index c(a asc, b desc));
CREATE TABLE
root@:26257/t> show create table ax;
+-------+-------------------------------------+
| Table |             CreateTable             |
+-------+-------------------------------------+
| ax    | CREATE TABLE ax (␤                  |
|       |     a INT NULL,␤                    |
|       |     b INT NULL,␤                    |
|       |     INDEX c (a, b),␤                |
|       |     FAMILY "primary" (a, b, rowid)␤ |
|       | )                                   |
+-------+-------------------------------------+
(1 row)
```

(Note: there are two things in the index column definitions that could be reported: direction, but also collation.)

### PR DESCRIPTION
**Pull request by madelynnblue**
Fixes #13923

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/13930)

### CODE REVIEW

File: pkg/sql/show.go [hunk 1]
```diff
@@ -414,6 +414,20 @@ func (p *planner) showCreateTable(tn parser.Name, desc *sqlbase.TableDescriptor)
...
 
+func makeIndexColNames(d sqlbase.IndexDescriptor) string {
+	var buf bytes.Buffer
+	for i, name := range d.ColumnNames {
+		if i > 0 {
+			buf.WriteString(", ")
+		}
+		buf.WriteString(parser.Name(name).String())
+		if dir := d.ColumnDirections[i]; dir != 0 {
+			buf.WriteString(" " + dir.String())
```
Comment (knz) [inline pkg/sql/show.go]: Either `fmt.Fprintf(" %s", dir)` or `buf.WriteByte(' '); buf.WriteString(dir.String())`, the string concatenation is not warranted.

File: pkg/sql/show.go [hunk 2]
```diff
@@ -414,6 +414,20 @@ func (p *planner) showCreateTable(tn parser.Name, desc *sqlbase.TableDescriptor)
...
 }
 
+func makeIndexColNames(d sqlbase.IndexDescriptor) string {
+	var buf bytes.Buffer
+	for i, name := range d.ColumnNames {
+		if i > 0 {
+			buf.WriteString(", ")
+		}
+		buf.WriteString(parser.Name(name).String())
+		if dir := d.ColumnDirections[i]; dir != 0 {
```
Comment (knz) [inline pkg/sql/show.go]: I would suggest to either

- drop the conditional on `dir != 0` and print the direction always, or
- print `ASC` always for direction 0.

**Comment by mjibson** [comment]
Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/sql/show.go, line 424 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13930#-KeanQUz1YWo9VRMttfV-r1-424:-Keat0eu7vjQYdHB2SnU:b-maysyb) ([raw file](https://github.com/cockroachdb/cockroach/blob/97ce8e8dfeb370b54d36011dcc87355fd131b391/pkg/sql/show.go#L424)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

I would suggest to either

- drop the conditional on `dir != 0` and print the direction always, or
- print `ASC` always for direction 0. 
</blockquote></details>

Printing the direction always is problematic because you can't (until that other PR you listed) list directions, for primary keys. I've added a flag to only print directions for non-primary indexes.

---

*[pkg/sql/show.go, line 425 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13930#-KeanQUz1YWo9VRMttfV-r1-425:-KeaupIh3DquO8yhJMU6:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/97ce8e8dfeb370b54d36011dcc87355fd131b391/pkg/sql/show.go#L425)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Either `fmt.Fprintf(&buf, " %s", dir)` or `buf.WriteByte(' '); buf.WriteString(dir.String())`, the string concatenation is not warranted.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13930)*

**Comment by knz** [comment]
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/sql/show.go, line 424 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13930#-KeanQUz1YWo9VRMttfV-r1-424:-Kecw6-ih2YE5ROFf_sx:b91flhs) ([raw file](https://github.com/cockroachdb/cockroach/blob/97ce8e8dfeb370b54d36011dcc87355fd131b391/pkg/sql/show.go#L424)):*
<details><summary><i>Previously, mjibson (Matt Jibson) wrote…</i></summary><blockquote>

Printing the direction always is problematic because you can't (until that other PR you listed) list directions, for primary keys. I've added a flag to only print directions for non-primary indexes.
</blockquote></details>

That other PR is merged now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13930)*

**Comment by mjibson** [comment]
RFAL. Changes from the other PR have been integrated.

**Comment by knz** [comment]
:lgtm:

---

Reviewed 6 of 6 files at r3.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13930#-:-Kee3TlZkipFKnzffIWi:bnfp4nl)*

**Pull request merged**